### PR TITLE
[stdlib] Audit use of @_transparent in FP code

### PR DESCRIPTION
**Pull request by moiseev**
<rdar://problem/45140678>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci please smoke test compiler performance

File: stdlib/public/core/FloatingPoint.swift [hunk 1]
```diff
@@ -1653,7 +1654,7 @@ extension FloatingPoint {
   ///
   /// - Parameter rule: The rounding rule to use.
   /// - Returns: The integral value found by rounding using `rule`.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPoint.swift]: This should almost surely stay transparent, as we want them to a single inline instruction in all circumstances on x86 or arm64.

File: stdlib/public/core/FloatingPoint.swift [hunk 2]
```diff
@@ -1682,7 +1683,7 @@ extension FloatingPoint {
   ///
   /// - Returns: The nearest integral value, or, if two integral values are
   ///   equally close, the integral value with greater magnitude.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPoint.swift]: Ditto here.

File: stdlib/public/core/FloatingPoint.swift [hunk 3]
```diff
@@ -1706,7 +1707,7 @@ extension FloatingPoint {
   ///
   /// To specify an alternative rule for rounding, use the `round(_:)` method
   /// instead.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPoint.swift]: and here.

File: stdlib/public/core/FloatingPoint.swift [hunk 4]
```diff
@@ -1817,7 +1820,7 @@ extension FloatingPoint {
   ///     // distance == 5.0
   ///
   /// - Returns: The square root of the value.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPoint.swift]: This too.

File: stdlib/public/core/FloatingPoint.swift [hunk 5]
```diff
@@ -1837,7 +1840,7 @@ extension FloatingPoint {
   ///   - lhs: One of the values to multiply before adding to this value.
   ///   - rhs: The other value to multiply.
   /// - Returns: The product of `lhs` and `rhs`, added to this value.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPoint.swift]: And this.

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 6]
```diff
@@ -995,7 +1000,7 @@ extension ${Self}: BinaryFloatingPoint {
   ///     // w1 == 7.0
   ///
   /// - Parameter rule: The rounding rule to use.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: and here.

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 7]
```diff
@@ -1125,14 +1130,14 @@ extension ${Self}: BinaryFloatingPoint {
...
   public mutating func formTruncatingRemainder(dividingBy other: ${Self}) {
     _value = Builtin.frem_FPIEEE${bits}(self._value, other._value)
   }
 
   /// Replaces this value with its square root, rounded to a representable
   /// value.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: and here.

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 8]
```diff
@@ -1143,7 +1148,7 @@ extension ${Self}: BinaryFloatingPoint {
   /// - Parameters:
   ///   - lhs: One of the values to multiply before adding to this value.
   ///   - rhs: The other value to multiply.
-  @_transparent
+  @inline(__always)
```
Comment (stephentyrone) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: and here.

File: stdlib/public/core/FloatingPoint.swift (hunk 1 above)
Comment (moiseev) [reply stdlib/public/core/FloatingPoint.swift]: Will it not with just `@inline(__always)`?

Comment (stephentyrone) [reply stdlib/public/core/FloatingPoint.swift]: If so, what does @_transparent mean, and why do we use it for arithmetic operators?

Comment (moiseev) [reply stdlib/public/core/FloatingPoint.swift]: I guess, this (👆) is a question for someone from @eeckstein's team.

**Comment by moiseev** [comment]
@swift-ci Please benchmark

File: stdlib/public/core/FloatingPoint.swift (hunk 1 above)
Comment (eeckstein) [reply stdlib/public/core/FloatingPoint.swift]: inline(__always) is a hint for the optimizer.
But transparent is also considered in Onone, mainly to get e.g. all the overflow diagnostics, which only work if arithmetic operations are inlined.
Also, transparent functions do not show up as function calls when debugging.

Comment (moiseev) [reply stdlib/public/core/FloatingPoint.swift]: Ah... The `Onone` argument is a good one. I did not think about that. Thanks!

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by moiseev** [comment]
@swift-ci Please clean smoke test Linux platform

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by airspeedswift** [comment]
GTM?

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Pull request merged**